src/ddsrt/include/dds/ddsrt/circlist.h

```diff
@@ -22,22 +26,95 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Macro to get the pointer to the user node, see @ref ddsrt_circlist_elem for an example.
+ */
 #define DDSRT_FROM_CIRCLIST(typ_, member_, cle_) ((typ_ *) ((char *) (cle_) - offsetof (typ_, member_)))
 
+/** @brief the circular doubly linked list */
 struct ddsrt_circlist {
-  struct ddsrt_circlist_elem *latest; /* pointer to latest inserted element */
+  struct ddsrt_circlist_elem *latest; /**< pointer to latest inserted element */
 };
 
+/**
+ * @brief The circlist element is the basic element of circular doubly linked list structure.
+ * 
+ * To store user data in the list, the element must be embedded in a user element, which is a struct containing the user data.
+ * All API calls with an element as input or output, deal with the circlist element. Example:
+ * @code{.c}
+ * typedef struct num_s{ // user element
```

Comment:
This a violation of MISRA rule 3.1 that @NoxPardalis just fixed elsewhere. For the comparable cases, he changed the documentation comment to use `///` instead of `/**` ... `*/`. I think that would be a practical solution here, too.